python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,576 @@
+                            field.getName(), column))
+                else:
+                    yield item
+
+    def _ingest(self, conn, cat, tableName, maxQueryLen):
+        """Ingest a BaseCatalog by converting it to one or more (large) INSERT or
+        REPLACE statements, executing those statements, and committing the result.
+        """
+        sqlPrefix = "REPLACE" if self.config.allowReplace else "INSERT"
+        sqlPrefix += " INTO {} (".format(tableName)
+        keys = []
+        columnNames = []
+        for item in self._schemaItems(cat.schema):
+            keys.append((item.key, fieldFormatters[item.field.getTypeString()]))
+            columnNames.append(self._columnName(item.field.getName()))
+        sqlPrefix += ",".join(columnNames)
+        sqlPrefix += ") VALUES "
+        pos = 0
+        while pos < len(cat):
+            sql = sqlPrefix
+            initialPos = pos
+            maxValueLen = maxQueryLen - len(sql)
+            while pos < len(cat):
+                row = cat[pos]
+                value = "("
+                value += ",".join([f.formatValue(row.get(k)) for (k, f) in keys])
+                value += "),"
+                maxValueLen -= len(value)
+                if maxValueLen < 0:
+                    break
+                else:
+                    sql += value
+                    pos += 1
+            if pos == initialPos:
+                # Have not made progress
+                raise RuntimeError("Single row is too large to insert")
+            self._executeSql(conn, sql[:-1])
+        conn.commit()
+
+    def _columnName(self, fieldName):
+        """Return the SQL column name for a given afw table field.
+        """
+        name = canonicalizeFieldName(fieldName)
+        return self.config.remap.get(name, name)
+
+    def _columnDef(self, field):
+        """Return the SQL column definition for a given afw table field.
+        """
+        sqlType = fieldFormatters[field.getTypeString()].sqlType(field)
+        return self._columnName(field.getName()) + " " + sqlType
+
+    def _createTable(self, conn, tableName, schema):
+        """Create a table using column definitions based on the provided afw
+        table schema, adding in any extra columns specified in the config.  If
+        one exists, the unique id column is given a key.
+        """
+        fields = [item.field for item in self._schemaItems(schema)]
+        names = [self._columnName(field.getName()) for field in fields]
+        if len(names) != len(set(n.lower() for n in names)):
+            raise RuntimeError(
+                "Schema contains columns that differ only by non-word characters "
+                "and/or case. In the database, these cannot be distinguished and "
+                "hence result in column name duplicates. Use the remap "
+                "configuration parameter to resolve this ambiguity.")
+        sql = "CREATE TABLE IF NOT EXISTS {} (\n\t".format(tableName)
```

Comment:
Then the CREATE statement does nothing, and inserts will append to the preexisting table if the schema is as expected or fail if not.